stdlib/public/core/StaticString.swift

```diff
@@ -19,11 +19,55 @@
 
 /// A string type designed to represent text that is known at compile time.
 ///
-/// Instances of the `StaticString` type are immutable. `StaticString` provides
-/// limited, pointer-based access to its contents, unlike Swift's more
-/// commonly used `String` type. A static string can store its value as a
-/// pointer to an ASCII code unit sequence, as a pointer to a UTF-8 code unit
-/// sequence, or as a single Unicode scalar value.
+/// Instances of the `StaticString` type are immutable.
+///
+/// `StaticString` provides only low-level access to its contents, unlike
+/// Swift's more commonly used `String` type.
+///
+/// A static string can store:
+///
+/// * either a pointer to a null-terminated sequence of UTF-8 code units;
+///
+///       let emoji: StaticString = "\u{1F600}"
+///       emoji.hasPointerRepresentation  //-> true
+///       emoji.isASCII                   //-> false
+///       emoji.unicodeScalar             //-> Fatal error!
+///       emoji.utf8CodeUnitCount         //-> 4
+///       emoji.utf8Start[0]              //-> 0xF0
+///       emoji.utf8Start[1]              //-> 0x9F
+///       emoji.utf8Start[2]              //-> 0x98
+///       emoji.utf8Start[3]              //-> 0x80
+///       emoji.utf8Start[4]              //-> 0x00
+///
+/// * or (under very limited circumstances) a single Unicode scalar value.
+///
+///       struct MyStaticScalar: ExpressibleByUnicodeScalarLiteral {
+///         typealias UnicodeScalarLiteralType = StaticString
+///         let value: StaticString
+///         init(unicodeScalarLiteral value: StaticString) {
+///           self.value = value
+///         }
+///       }
+///
+///       let emoji: StaticString = MyStaticScalar("\u{1F600}").value
+///       emoji.hasPointerRepresentation  //-> false
+///       emoji.isASCII                   //-> false
+///       emoji.unicodeScalar.value       //-> 0x1F600
+///       emoji.utf8CodeUnitCount         //-> Fatal error!
+///       emoji.utf8Start                 //-> Fatal error!
+///
+/// The `withUTF8Buffer(_:)` method can be used to access the contents,
+/// regardless of whether the static string has a pointer representation.
```

Comment:
Let's use active voice here:
```suggestion
/// You can use the `withUTF8Buffer(_:)` method to access a static string's
/// contents, regardless of which representation the static string uses.
```